sml/src/sml_number.c

```diff
@@ -24,8 +24,25 @@
 #define SML_BIG_ENDIAN 1
 #define SML_LITTLE_ENDIAN 0
 
-int sml_number_endian();
-void sml_number_byte_swap(unsigned char *bytes, int bytes_len);
+static void sml_number_byte_swap(unsigned char *bytes, int bytes_len) {
+	int i;
+	unsigned char ob[bytes_len];
```

Comment:
that's a bit risky as bytes_len can be huge. (but it was part of orig. change already)